### PR TITLE
Add a way to accept all changes after multiple successful SaveChanges.

### LINKED ISSUE #744: Integrate updating entity state and the committing of generated values into entities with user-controlled transactions
Values generated in the update pipeline are saved in sidecars so that if the save fails these values can be discarded. Currently the values are always committed from sidecars into the actual entity instances when SaveChanges ends. But if the user is handling transactions manually then this commit should not happen until the transaction is committed.

Likewise, entity state is updated at the end of SaveChanges, but again this should only be done when the transaction is actually committed--this is the AcceptChanges functionality in EF6.

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #744

### CODE REVIEW

File: src/EntityFramework.Core/DbContext.cs [hunk 1]
```diff
@@ -290,13 +290,34 @@ protected internal virtual void OnModelCreating(ModelBuilder modelBuilder)
...
+        /// <remarks>
+        ///     This method will automatically call <see cref="ChangeTracker.DetectChanges" /> to discover any changes
+        ///     to entity instances before saving to the underlying data store. This can be disabled via
+        ///     <see cref="ChangeTracker.AutoDetectChangesEnabled" />.
+        /// </remarks>
+        /// <returns>
+        ///     The number of state entries written to the underlying data store.
+        /// </returns>
+        [DebuggerStepThrough]
+        public virtual int SaveChanges(bool acceptAllChangesOnSuccess)
```
Comment (ajcvickers) [inline src/EntityFramework.Core/DbContext.cs]: Default value rather than new overload? Not sure which is best...

Comment (ajcvickers) [inline src/EntityFramework.Core/DbContext.cs]: Never mind. CancellationToken on the async version makes the default value approach less nice.

**Comment by ajcvickers** [comment]
:shipit:

File: src/EntityFramework.Core/DbContext.cs (hunk 1 above)
Comment (AndriySvyryd) [inline src/EntityFramework.Core/DbContext.cs]: Default value means that we will never change it.

**Pull request merged**